api/deployments/component_handler.go

```diff
@@ -169,7 +172,12 @@ func GetComponentStateFromSpec(
 		}
 		componentPodNames = getPodNames(componentPods)
 		environmentVariables = getRadixEnvironmentVariables(componentPods)
-		replicaSummaryList = getReplicaSummaryList(componentPods)
+		eventList, err := kubequery.GetEventsForEnvironment(ctx, kubeClient, appName, deployment.Environment)
+		if err != nil {
+			return nil, err
+		}
+		lastEventWarnings := event.ConvertToEventWarnings(eventList)
+		replicaSummaryList = getReplicaSummaryList(componentPods, lastEventWarnings)
```

Comment:
Not sure how useful it is to include the last warning event. There can be multiple warning events, and only showing the last will potentially hide other important events. Perhaps a better approach is to have an endpoint for the component and/or replica that returns events for the particular object. We can then show it in web console as an event list, similar to the event list on the environment page

Reply:
Alternative can be to show events within the replica page, but it will not show the rootcause of some issues with replica after it was deleted